### PR TITLE
completed batch creator implementation

### PR DESCRIPTION
**Pull request by SrujithPoondla**
-- Removed sample.py as it is not used 
-- Replaced Arrays with Queues
-- Implemented Batch Creator
-- Added an event to load.py to stop the batch creation and samples creation process

### CODE REVIEW

File: sampling/batch_creator.py [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+
+    def run(self):
+        """
+        Keep creating the batches in the background
+        """
+        while not self.stop_batch_creator.is_set():
+            if self.ds.samples.empty():
+                print("Waiting for sample creator to create a sample")
+                continue
+            elif self.batches is self.max_batches:
```
Comment (aarati-K) [inline sampling/batch_creator.py]: `elif self.batches.full():`

Comment (SrujithPoondla) [reply sampling/batch_creator.py]: Sure, will change that

File: sampling/batch_creator.py [hunk 2]
```diff
@@ -1,14 +1,39 @@
...
+                continue
+            else:
+                i=0
+                curr_sample = self.ds.samples.get()
+                curr_batch = []
+                while i < self.batch_size:
+                    curr_batch.append(curr_sample[i+self.offset])
+                    i = i + 1
+                self.offset += i
+                self.batches.put(np.array(curr_batch))
```
Comment (aarati-K) [inline sampling/batch_creator.py:43]: What if self.batches becomes full while putting batches?

Comment (SrujithPoondla) [reply sampling/batch_creator.py:43]: The second condition is looking for that right?

File: sampling/batch_creator.py [hunk 3]
```diff
@@ -1,14 +1,39 @@
...
+        while not self.stop_batch_creator.is_set():
+            if self.ds.samples.empty():
+                print("Waiting for sample creator to create a sample")
+                continue
+            elif self.batches is self.max_batches:
+                print("Waiting for client to take a batch from batches array")
+                continue
+            else:
+                i=0
+                curr_sample = self.ds.samples.get()
```
Comment (aarati-K) [inline sampling/batch_creator.py]: Follow up to the comment below. You could alternately do:
`if not cur_sample or self.offset == self.ds.sample_size:`
`    cur_sample = self.ds.samples.get()`

File: sampling/batch_creator.py (hunk 2 above)
Comment (aarati-K) [reply sampling/batch_creator.py:43]: But you are pulling a new sample from `ds.samples` queue for every batch.

Comment (SrujithPoondla) [reply sampling/batch_creator.py:43]: Okay i get it, will change that

File: sampling/batch_creator.py [hunk 4]
```diff
@@ -1,14 +1,45 @@
...
+        """
+        Keep creating the batches in the background
+        """
+        while not self.stop_batch_creator.is_set():
+            if self.ds.samples.empty():
+                print("Waiting for sample creator to create a sample")
+                continue
+            elif self.batches.full():
+                print("Waiting for client to take a batch from batches array")
+                sleep(10)
```
Comment (aarati-K) [inline sampling/batch_creator.py]: for 10 seconds?

File: sampling/batch_creator.py [hunk 5]
```diff
@@ -1,14 +1,45 @@
...
+                continue
+            else:
+                i=0
+                curr_sample =[]
```
Comment (aarati-K) [inline sampling/batch_creator.py]: Why do you have two variables `cur_sample` and `curr_sample`?

Comment (SrujithPoondla) [reply sampling/batch_creator.py]: :| fixed

File: load/load.py [hunk 6]
```diff
@@ -1,32 +1,34 @@
...
+    def __init__(self, data_store, batch_size=10):
+        """
+          Start the batchCreator and sampleCreator.
+          Read the memory config file, and create the right number
+          of processes.
+        """
+        self.data_store = data_store
+        self.data_store.batch_size = batch_size
+        # Event to stop batch creator and sample creator
+        self.event = Event()
```
Comment (aarati-K) [inline load/load.py:17]: Can you also add a function to set this event?

Comment (SrujithPoondla) [reply load/load.py:17]: Yup, added. It will be there in next push

File: sampling/batch_creator.py (hunk 5 above)
Comment (aarati-K) [reply sampling/batch_creator.py]: No there are still two separate variables.

Comment (SrujithPoondla) [reply sampling/batch_creator.py]: I just fixed it. I didn't push the fixed code yet.

Comment (aarati-K) [reply sampling/batch_creator.py]: Okay maybe you haven't pushed.

File: sampling/batch_creator.py [hunk 7]
```diff
@@ -1,14 +1,44 @@
...
+            if self.ds.samples.empty():
+                print("Waiting for sample creator to create a sample")
+                continue
+            elif self.batches.full():
+                print("Waiting for client to take a batch from batches array")
+                continue
+            else:
+                i = 0
+                curr_sample = []
+                if not curr_sample or self.offset == self.ds.sample_size:
```
Comment (aarati-K) [inline sampling/batch_creator.py:35]: You are setting curr_sample to [] right above this if statement. This if condition will always hold true.

**Pull request merged**